### PR TITLE
docs: add docstring

### PR DESCRIPTION
**Pull request by jokasimr**

### CODE REVIEW

File: src/scippneutron/absorption/base.py [hunk 1]
```diff
@@ -15,6 +15,33 @@ def compute_transmission_map(
...
+    beam_direction:
+        direction of the incoming beam
+    wavelength:
+        an array of wavelengths for which to compute the transmission fraction
+    detector_position:
+        an array of vectors where to compute the transmission fraction
+    quadrature_kind:
+        What kind of quadrature to use.
+        A denser quadrature makes the result more accurate but takes longer to compute.
+        What options exists depend on the sample shape.
```
Comment (jl-wynen) [inline src/scippneutron/absorption/base.py:38]: You can use a `Literal` as the type of `quadrature_kind` to tell the users which arguments are possible. As it stands, it is very difficult to find that out.

File: src/scippneutron/absorption/base.py [hunk 2]
```diff
@@ -15,6 +15,33 @@ def compute_transmission_map(
...
+
+    Returns
+    -------
+        the transmission fraction as a function of detector_position and wavelength
```
Comment (jl-wynen) [inline src/scippneutron/absorption/base.py:42]:
```suggestion
    :
        the transmission fraction as a function of detector_position and wavelength
```
To make Sphinx render it properly and show the return type. See also https://scipp.github.io/scippneutron/developer/coding-conventions.html#docstring-format

File: src/scippneutron/absorption/base.py [hunk 3]
```diff
@@ -15,6 +15,33 @@ def compute_transmission_map(
...
     quadrature_kind: Any = 'medium',
 ) -> sc.DataArray:
+    """
+    Computes the fraction of neutrons transmitted through the sample
+    given that they scattered to the positions in the ``detector_positions`` array.
+
+    Parameters
+    ----------
+    sample_shape:
+        the size and shape of the sample
```
Comment (jl-wynen) [inline src/scippneutron/absorption/base.py:25]: Capitalization and punctuation are inconsistent here. Please make every entry in Parameters (and the return documentation) one or more sentences. I.e., begin with a capital letter and end with a period.

File: src/scippneutron/absorption/base.py [hunk 4]
```diff
@@ -15,6 +15,33 @@ def compute_transmission_map(
     detector_position: sc.Variable,
     quadrature_kind: Any = 'medium',
 ) -> sc.DataArray:
+    """
+    Computes the fraction of neutrons transmitted through the sample
+    given that they scattered to the positions in the ``detector_positions`` array.
```
Comment (jl-wynen) [inline src/scippneutron/absorption/base.py:20]: The first sentence of a docstring should only be a single line and it should be its own paragraph, i.e., separated by an empty line from the rest. (See also https://peps.python.org/pep-0257/#multi-line-docstrings) Sphinx rtelies on this and shows the first paragraph in summary tables. And there is little space there. So please split it up and write a concise summary plus an expanded explanation.

Typically, the summary uses imperative mood. But we don't do that consistently in Scipp.

**Review by jl-wynen — COMMENTED** [review]
Looks good, but I take this opportunity to be pedantic about style, generally following [PEP257](https://peps.python.org/pep-0257) and [pydocstyle / ruff](https://docs.astral.sh/ruff/rules/#pydocstyle-d)

**Comment by jokasimr** [comment]
> Looks good, but I take this opportunity to be pedantic about style

Please be!

**Review by jl-wynen — APPROVED** [review]

**Pull request merged**